qiskit/circuit/library/evolution/suzuki_trotter.py

```diff
@@ -0,0 +1,102 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Lie-Trotter product formula."""
+
+from typing import List, Callable, Optional, Union
+import numpy as np
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.quantum_info.operators import SparsePauliOp, Pauli
+
+from .product_formula import ProductFormula
+
+
+class SuzukiTrotter(ProductFormula):
+    """The (higher order) Suzuki-Trotter product formula."""
+
+    def __init__(
+        self,
+        reps: int = 1,
+        order: int = 2,
+        atomic_evolution: Optional[Callable[[SparsePauliOp, float], QuantumCircuit]] = None,
+        insert_barriers: bool = False,
+        cx_structure: str = "chain",
+    ) -> None:
+        """
+        Args:
+            order: The order of the product formula.
+            reps: The number of time steps.
+            atomic_evolution: A function to construct the circuit for the evolution of single operators.
```

Comment:
Missing some args - maybe `order` and `reps` would be preferable in same order as init ie reps then order...